features/0023-connection-initiate/README.md

```diff
@@ -0,0 +1,425 @@
+
+**request_processing_error** - This error is sent when the inviter was processing the request with the intent to accept the request, but some processing error occurred. This error indicates that the request should be resent as-is.
+
+**response_not_accepted** - The error indicates that the response has been rejected for a reason listed in the error_report. Typical reasons include not accepting the method of the provided DID, unknown endpoint protocols, invalid signature, etc. The response can be resent _after_ the appropriate corrections have been made.
+
+**response_processing_error** - This error is sent when the invitee was processing the response with the intent to accept the response, but some processing error occurred. This error indicates that the response should be resent as-is.
+
+No errors are sent in timeout situations. If the inviter or invitee wishes to retract the messages they sent, they record so locally and return a `request_not_accepted` or `response_not_accepted` error when the other party sends a request or response .
+
+#### Error Message Example
+
+```
+{
+  "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/problem_report",
+  "@id": "5678876542345",
+  "~thread": { "thid": "<@id of message related to problem>" },
+  "~i10n": { "locale": "en"},
+  "problem-code": "request_not_accepted", // matches codes listed above
+  "explain": "Unsupported DID method for provided DID."
+}
+```
+
+#### Error Message Attributes
+
+- The `@type` attribute is a required string value that denotes that the received message is a problem_report within the connections family.
+- The `~thread` attribute provides a context for the problem, referring to the message which contains the problem.
+- Use of `~i10n` is encouraged, with at least locale defined for the message.
+- The `problem-code` attribute contains one of a fixed set of codes defined in the list above.
+- The `explain` attribute contains a human readable message which indicates the problem.
+
+### Flow Overview
+The _inviter_ gives provisional connection information to the _invitee_. 
+The _invitee_ uses provisional information to send a DID and DID Doc to the _inviter_.
+The _inviter_ uses sent DID Doc information to send a DID and DID Doc to the _invitee_.
+The *invitee* sends the *inviter* an ack or any other message that confirms the response was received.
+
+## 0. Invitation to Connect
+[0-invitation]: #1-invitation
+
+An invitation to connect may be transferred using any method that can reliably transmit text. The result  must be the essential data necessary to initiate a [Connection Request](#2.-connection-request) message. A connection invitation is an agent message with agent plaintext format, but is an **out-of-band communication** and therefore not communicated using wire level encoding or encryption. The necessary data that an invitation to connect must result in is:
+* suggested label
+
+* publicly resolvable did
+
+  OR
+
+* suggested label
+
+* recipientKeys
```

Comment:
@TelegramSam these three can be substituted by the [`~service`](https://github.com/hyperledger/aries-rfcs/tree/master/features/0056-service-decorator) decorator